src/test_suite/test_runner_main.ts

```diff
@@ -162,155 +203,208 @@ export async function testRunnerMain(
         console.log('Running all tests.');
     }
 
-    let suite: TestSuite;
-    try {
-        suite = TestSuite.fromYamlString(fs.readFileSync(testFile, 'utf8'));
-    } catch (err) {
-        if (err.code === 'ENOENT' || err.code === 'EISDIR') {
-            const message = `Cannot open test file "${err.path}"`;
-            fail(message);
-            // Unreachable code exists for type assertion.
-            suite = undefined!;
-        } else {
-            throw err;
+    const testSuites: TestSuite[] = [];
+    const allResults = new AggregatedResults();
+
+    for (const testFile of testFiles) {
+        let suite: TestSuite;
+        try {
+            suite = TestSuite.fromYamlString(fs.readFileSync(testFile, 'utf8'));
+        } catch (err) {
+            if (err.code === 'ENOENT' || err.code === 'EISDIR') {
+                const message = `Cannot open test file "${err.path}"`;
+                fail(message);
+                // Unreachable code exists for type assertion.
+                suite = undefined!;
+            } else {
+                throw err;
+            }
         }
-    }
 
-    if (runOneTest !== undefined) {
-        if (runOneTest >= 0 && runOneTest < suite.tests.length) {
-            suite = new TestSuite([suite.tests[runOneTest]]);
-        } else {
-            const message = `Invalid test number ${runOneTest}`;
-            fail(message);
+        if (runOneTest !== undefined) {
+            if (runOneTest >= 0 && runOneTest < suite.tests.length) {
+                suite = new TestSuite([suite.tests[runOneTest]]);
+            } else {
+                const message = `Invalid test number ${runOneTest}`;
+                fail(message);
+            }
         }
+
+        testSuites.push(suite);
     }
+
     if (brief) {
         console.log(' ');
         console.log('Displaying test utterances without running.');
-        for (const test of suite.filteredTests(suiteExpression)) {
-            console.log(`Test ${test.id}: ${test.comment}`);
-            for (const step of test.steps) {
-                const input = getYamlInputText(step, correctionLevel);
-                console.log(`  ${input}`);
+        testSuites.forEach(suite => {
```

Comment:
Not a huge proponent of using `.forEach` personally (happy to discuss why, lots of articles out there too). Mainly though it's inconsistent with the rest of the code base which uses `for of` already.

Reply:
I agree with Drew on this one.

Reply:
`for of` it is then :)